src/box/vy_log.c

```diff
@@ -1026,6 +1026,45 @@ vy_log_bootstrap(void)
 	return 0;
 }
 
+/**
+ * Return true if the last vylog is new and contains no written rows
+ * (i.e. last entry is VY_LOG_SNAPSHOT).
+ * In case of any errors log them and return false.
+ */
+static bool
+vy_log_last_is_empty(void)
+{
+	const char *path =
+		vy_log_filename(vclock_sum(&vy_log.last_checkpoint));
+	if (access(path, F_OK) < 0) {
+		say_error("Failed to access last vylog");
+		return false;
+	}
+	struct xlog_cursor cursor;
+	if (xdir_open_cursor(&vy_log.dir,
+			     vclock_sum(&vy_log.last_checkpoint),
+			     &cursor) < 0) {
+		diag_log();
+		diag_clear(diag_get());
+		return false;
+	}
+	struct xrow_header row;
+	while ((xlog_cursor_next(&cursor, &row, false) == 0)) {
+		struct vy_log_record record;
+		if (vy_log_record_decode(&record, &row) != 0) {
+			diag_log();
+			diag_clear(diag_get());
+			return false;
+		}
+		if (record.type == VY_LOG_SNAPSHOT) {
+			if (xlog_cursor_next(&cursor, &row, false) != 1)
+				return false;
+			return true;
+		}
+	}
+	return false;
```

Comment:
I guess in case when `xlog_cursor_next` returned -1 we should also clear the diag.